docs/decisions/0049-agents-assistantsV2.md

```diff
@@ -0,0 +1,157 @@
+
+# Agent Framework - Assistant V2 Migration
+
+## Context and Problem Statement
+
+Open AI has release the _Assistants V2_ API.  This builds on top of the V1 _assistant_ concept, but also invalidates certain V1 features.  In addition, the _dotnet_ API that supports _Assistant V2_ features is entirely divergent on the `Azure.AI.OpenAI.Assistants` SDK that is currently in use.
+
+### Open Issues
+- **Streaming:** To be addressed as a discrete feature
+
+
+## Design
+
+Migrating to Assistant V2 API is a breaking change to the existing package due to:
+- Class definition changes
+- Underlying capability differences (e.g. `file-search` vs `retrieval`)
+- Underlying V2 SDK is version incompatible with V1 (`OpenAI` and `Azure.AI.OpenAI`)
+
+
+### Configuration Classes
+
+Specific configuration/setttings classes are introduced to support the ability to define assistant behavior at each of the supported articulation points (i.e. _assistant_, _thread_, & _run_).
+
+
+#### Assistant Definition
+
+The `OpenAIAssistantDefinition` was previously used only when enumerating a list of stored agents.  It has been evolved to also be used as input for creating and agent and exposed as a discrete property on the `OpenAIAssistantAgent` instance.
+
+This includes optional _execution_settings_ for defining default _run_ behavior.  Since these execution settings are not part of the remote assistant definition, they are persisted in the assistant metadata for when an existing agent is retrieved.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-definition.png"  style="width: 360pt;"></kbd>
+</p>
+
+
+#### Assistant Invocation Settings
+
+When invoking an `OpenAIAssistantAgent` directly (no-chat), settings that only apply to a discrete run may be specified.  These settings are defined as `OpenAIAssistantInvocationSettings`.
+
+<p align="center">
+<kbd><img src="diagrams/assistant-invocationsettings.png" style="width: 220pt;"></kbd>
+</p>
+
+
+#### Thread Creation Settings
+
+When invoking an `OpenAIAssistantAgent` directly (no-chat), a thread must be explicitly managed.  When doing so, thread specific settings may be specified.  These settings are defined as `OpenAIThreadCreationSettings`.
```

Comment:
Is it necessary to have `EnableCodeInterpreter` in both these settings and `OpenAIAssistantInvocationSettings`? Which would have precedence?

Reply:
The thread-settings have precedence over the assistant settings (when specified).

The invokcation-settings have precedence over all setting (when specified).  

Funny, it just occured to me to clarify this in the ADR...good call-out.